src/lib.rs

```diff
@@ -435,224 +437,319 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
+
+                                let delta = pointer_pos - start;
+                                if delta.x.abs() > 30.0 || delta.y.abs() > 6.0 {
+                                    ui.ctx().translate_layer(layer_id, delta);
+
+                                    drag_data = Some((node_index, tab_index));
+                                }
                             }
 
-                            let is_active = *active == tab_index || is_being_dragged;
-                            let label = tab_viewer.title(tab);
-
-                            let response = if is_being_dragged {
-                                let layer_id = LayerId::new(Order::Tooltip, id);
-                                let mut response = ui
-                                    .with_layer_id(layer_id, |ui| {
-                                        style.tab_title(
-                                            ui,
-                                            label,
-                                            is_active,
-                                            is_active && Some(node_index) == focused,
-                                            is_being_dragged,
-                                            id,
-                                            expanded_width,
-                                            self.show_close_buttons,
-                                        )
-                                    })
-                                    .response;
-
-                                let sense = Sense::click_and_drag();
-                                response = ui.interact(response.rect, id, sense);
-
-                                if let Some(pointer_pos) = ui.ctx().pointer_interact_pos() {
-                                    let center = response.rect.center();
-                                    let start = state.drag_start.unwrap_or(center);
-
-                                    let delta = pointer_pos - start;
-                                    if delta.x.abs() > 30.0 || delta.y.abs() > 6.0 {
-                                        ui.ctx().translate_layer(layer_id, delta);
-
-                                        drag_data = Some((node_index, tab_index));
-                                    }
-                                }
+                            response
+                        } else {
+                            let (mut response, close_response) = style.tab_title(
+                                ui,
+                                label,
+                                is_active && Some(node_index) == focused,
+                                is_active,
+                                is_being_dragged,
+                                id,
+                                expanded_width,
+                                self.show_close_buttons,
+                            );
+
+                            let (close_hovered, close_clicked) = match close_response {
```

Comment:
This can be replaced with:
```rs
let (close_hovered, close_clicked) = close_response
    .map(|res| (res.hovered(), res.clicked()))
    .unwrap_or_default();
```